importer/src/configuration.rs

```diff
@@ -0,0 +1,62 @@
+use serde::Deserialize;
+use std::fs;
+
+pub struct ImporterConfiguration {
+    pub curl: String,
+    pub osmconvert: String,
+    pub unzip: String,
+    pub gunzip: String,
+    pub gunzip_args: Option<String>,
+}
+
+#[derive(Deserialize)]
+struct RawImporterConfiguration {
+    pub curl: Option<String>,
+    pub osmconvert: Option<String>,
+    pub unzip: Option<String>,
+    pub gunzip: Option<String>,
+    pub gunzip_args: Option<String>,
+}
+
+pub fn load_configuration() -> ImporterConfiguration {
+
+    match fs::read_to_string("importer.toml") {
+        Ok(text) => {
+            match toml::from_str::<RawImporterConfiguration>(&text) {
```

Comment:
Could we leverage serde to express default values for the config in a simpler way? Some combination of https://serde.rs/attr-default.html, writing in JSON (should be just as easy to manually fill out), and `abstutil::maybe_read_json`. I think you can just keep `ImporterConfiguration`, using directives like `#[serde(default = "curl")]`.

Reply:
I wasn't very happy with this roundabout default mechanism. Unfortunately I think `#[serde(default = "curl")]` would expect a `curl` function to exist in scope that it can call to get a string. So it's not quite ideal. I can definitely change it to use json instead of toml.

Reply:
Ah you're right, I didn't realize it's so unergonomic... I guess `fn default_curl() -> String { "curl".to_string() }` and the 3 others is still a bit simpler than having two structs and manually doing `unwrap_or_else`-type behavior though.

Reply:
Ok, I completely missed the struct-level `#[serde(default)]` which just requires implementing the Default interface. So all the defaults are defined once in a `default` method instead of in individual methods, and serde handles the rest.